Add new SDK for Microsoft.MachineLearning/webservices RP

tudordid opened the pull request:
First iteration of the .NET sdk code for the MachineLearning/ webservices SDK

tudordid committed:
Initial commit to get repo in same state as Klein's fork.

tudordid committed:
Push latest form for generated code after swagger schema updates

tudordid committed:
Corrected property name to "value" in PaginatedWebServicesList contract

tudordid committed:
Added missing values to ProvisioningState enum

tudordid committed:
regenerated SDK code to include new changes made to the swagger schema
  - addition of type property to GraphParameter
  - removal of not needed API /subscriptions/{subscriptionId}/providers/Microsoft/MachineLearning/checkNameAvailability
 - fixes to documentation of classes

tudordid committed:
Added ParameterType class that was missed in previous commit

tudordid committed:
Updated UT to include new 'type' property for GraphParameter

tudordid committed:
Regenerated code from swagger spec

tudordid committed:
- regenerated code after swagger changes
- added utility to support serialization / deserialization of models
- refactored model tests

tudordid committed:
Regenerated SDK code after swagger schema updates

tudordid committed:
Added recorded UTs against dogfood (will re-record against Prod when possible)

tudordid committed:
Regenerated code after swagger update + pointed generation script ot preview spec

tudordid committed:
- regenerated SDK code from updated swagger spec
- added *Additions classes to generated code to inject correct request id in error responses for async operations (temp fix)

tudordid committed:
Created new test recordings against Dogfood after latest swagger schema changes

tudordid committed:
- regenerated SDK code from latest swagger spec (pushed to Azure repo)
- renamed projects to be ML generic, as we plan to host the workspaces SDK code here as well, not just web services
- changed test code to also create a commitment plan as part of test execution

tudordid committed:
- regenerated SDK classes off latest schema
  - renamed WebServicePropertiesForGraphPackage class to GraphPackage
  - made several properties required
- Re-recorded tests against Dogfood with dynamic creation of CP resource now + more cleanup

tudordid committed:
Regenerated SDK classes from latest swagger +re-recorded UTs

tudordid committed:
Recorded the tests against Prod + disposal code refactoring

tudordid committed:
fix path for MachineLearning project under global.json after  name refactoring

tudordid committed:
Set initial nuget version to 0.9.0-preview

azuresdkci commented:
Can one of the admins verify this patch?

azurecla commented:
Hi __@tudordid__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (tudordid). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

hovsepm commented:
@azuresdkci add to whitelist

hovsepm commented on src/ResourceManagement/MachineLearning/Microsoft.Azure.Management.MachineLearning/generate.cmd:
```diff
@@ -0,0 +1,17 @@
+@echo off
+set autoRestVersion=0.16.0-Nightly20160501
+if  "%1" == "" (
+    set specFile="https://raw.githubusercontent.com/tudordid/azure-rest-api-specs/master/arm-machinelearning/2016-05-01-preview/swagger/webservices.json"
```
should not be from your fork. You should generate the sources from Azure github repo

tudordid replied:
Correct, I forgot to update this after we have pushed the swagger spec in the main repo. Will fix.

hovsepm commented on src/ResourceManagement/MachineLearning/Microsoft.Azure.Management.MachineLearning/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,31 @@
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Machine Learning.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
this should be 0.9.0 (since your project.json specifies that version)

tudordid replied:
fixed

hovsepm commented on src/ResourceManagement/MachineLearning/Microsoft.Azure.Management.MachineLearning/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,31 @@
+[assembly: AssemblyTitle("Microsoft Azure Machine Learning Management Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Machine Learning.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
```
this should be 0.9.0 (since your project.json specifies that version)

hovsepm commented on src/ResourceManagement/MachineLearning/MachineLearning.Tests/ScenarioTests/WebServiceTests.cs:
```diff
@@ -0,0 +1,396 @@
+            ////Environment.SetEnvironmentVariable("TEST_CSM_ORGID_AUTHENTICATION", "SubscriptionId=7b373400-c82e-453b-a97b-c53e14325b8b;Environment=Dogfood;UserId=admin@aad216.ccsctp.net;Password=Pa$$w0rd");
+            ////Environment.SetEnvironmentVariable("TEST_CSM_ORGID_AUTHENTICATION", "SubscriptionId=80c77c76-74ba-4c8c-8229-4c3b2957990c;Environment=Prod");
+            ////Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Record");
+        }
```
remove these lines and change the password of your admin@aad216.ccsctp.net.

tudordid replied:
done

hovsepm commented on src/ResourceManagement/MachineLearning/MachineLearning.Tests/ScenarioTests/WebServiceTests.cs:
```diff
@@ -0,0 +1,396 @@
+                    Assert.Equal(serviceKeys.Primary, serviceUpdates.Properties.Keys.Primary);
+                    Assert.Equal(serviceKeys.Secondary, serviceDefinition.Properties.Keys.Secondary);
+
+                    
```
no need for extra line breaks.

hovsepm commented:
@tudordid you will need to pull from Azure fork AutoRest branch and resolve conflicts

tudordid committed:
Remove recording variables

tudordid committed:
Addressing pull request comments:
 - fixed generation script to use swager spec from Azure repo
 - changed AssemblyInfo.cs file to show correct assembly version
 - removed blank lines

tudordid committed:
fixes after rebase to resolve conflict with Azure origin
